feat(HRs): add automatic-updates-helmrelease guide (helm-releases section, page 3/4) (#3185)
giantswarm/docs: src/content/tutorials/continuous-deployment/helm-releases/automatic-updates-helmrelease.md

---

Flux can roll new chart versions for a HelmRelease without manual tag bumps. Two approaches:

1. **SemVer range on the `OCIRepository`** (recommended). Flux scans the OCI registry on its interval, picks the highest tag matching your range, and reconciles the HelmRelease. Nothing is committed back to Git: the chart selection is set by the `OCIRepository` resource alone.
2. **Image automation with Git commits.** Flux watches a registry for new tags, writes the selected tag back into your `ocirepository.yaml`, and commits the change. Adds an audit trail in Git at the cost of more moving parts.

The App CR equivalent of this guide is [Enable automatic updates in Apps]({{< relref "/tutorials/continuous-deployment/apps/automatic-updates-appcr" >}}).

## Pick up versions via OCIRepository range

Edit `ocirepository.yaml` and replace the `tag` field with a `semver` field that matches the versions you want Flux to pick up:

```yaml
apiVersion: source.toolkit.fluxcd.io/v1
kind: OCIRepository
metadata:
  name: my-app
  namespace: org-acmedev
spec:
  url: oci://gsoci.azurecr.io/charts/giantswarm/my-app
  ref:
    semver: ">=2.0.0 <3.0.0"   # was tag: "2.9.1"
  interval: 5m
```

> On each reconciliation interval (`5m` above), Flux fetches the tag list from the registry, picks the highest version satisfying the range, and updates the OCIRepository's status. The HelmRelease detects the chart revision change and reconciles a Helm upgrade.

Author: ```suggestion
At each reconciliation interval (`5m`), Flux fetches the tag list from the registry, selects the highest version within the range, and updates the OCIRepository's status. The HelmRelease detects changes to chart revisions and reconciles a Helm upgrade.
```
